token-swap/program/src/curve/offset.rs

```diff
@@ -0,0 +1,194 @@
+//! The Uniswap invariant calculator with an extra offset
```

Comment:
Can you provide more context on why this is useful? It seems like it makes it so you don't have to provide so much liquidity on one side of the pool and the offset effectively sets a minimum exchange price.

Reply:
You've got it exactly right -- it's specifically geared towards new exchange offerings that don't require tons of liquidity on the other side.  For example, if I want to sell 1 million of my new token, and I think each is worth 2 USDC, I make this pool with an offset of 2 million USDC, so I can start selling the token at that rate